components/chainSelect/ChainSelect.tsx

```diff
@@ -4,18 +4,40 @@ import { StargateClient } from "@cosmjs/stargate";
 
 import GearIcon from "../icons/Gear";
 import Button from "../inputs/Button";
-import Input from "../../components/inputs/Input";
+import Input from "../inputs/Input";
 import { useAppContext } from "../../context/AppContext";
 import Select from "../inputs/Select";
 import StackableContainer from "../layout/StackableContainer";
 
+interface ChainOption {
+  label: string;
+  value: number;
+}
+
+interface GithubChainRegistryItem {
+  name: string;
+  path: string;
+  sha: string;
+  size: number;
+  url: string;
+  html_url: string;
+  git_url: string;
+  download_url: string | null;
+  type: string;
+  _links: {
+    self: string;
+    git: string;
+    html: string;
+  };
+}
+
 const ChainSelect = () => {
   const { state, dispatch } = useAppContext();
 
   // UI State
   const [chainArray, setChainArray] = useState();
-  const [chainOptions, setChainOptions] = useState();
-  const [chainError, setChainError] = useState();
+  const [chainOptions, setChainOptions] = useState<ChainOption[]>([]);
+  const [chainError, setChainError] = useState<string | null>();
```

Comment:
Does this compile? What's the default value in this case?

Reply:
It does compile, default value is an empty array which leads to the Select UI to be empty until chain options are loaded from the registry.

Reply:
I mean this line with no default set:

```ts
  const [chainError, setChainError] = useState<string | null>();
```

Turns out the type becomes `string | null | undefined` and defaults to `undefined`.

I think there are two nice ways to fix:

1. Use `null`:   `const [chainError, setChainError] = useState<string | null>(null);`
2. Use `undefined`: `const [chainError, setChainError] = useState<string>();`

where 1. is probably what you wanted.

Reply:
ah right, it should be null.